extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -744,6 +766,40 @@ export class AmpA4A extends AMP.BaseElement {
   }
 
   /**
+   * Refreshes ad slot by fetching a new creative and rendering it. This leaves
+   * the current creative displayed until the next one is ready.
+   *
+   * @param {function()} restartRefresh When called, this function will restart
+   *   the refresh cycle.
+   */
+  refresh(restartRefresh) {
+    this.isRefreshing = true;
+    this.tearDownSlot();
+    this.initiateAdRequest();
+    const promiseId = this.promiseId_;
+    this.adPromise_.then(() => {
+      if (!this.isRefreshing || promiseId != this.promiseId_) {
+        // If this refresh cycle was canceled, such as in a no-content
+        // response case, keep showing the old creative.
+        restartRefresh();
+        return;
+      }
+      this.togglePlaceholder(true);
+      this.destroyFrame(true);
+      // We don't want the next creative to appear too suddenly, so we
+      // show the loader for a quarter of a second before switching to
+      // the new creative.
+      timerFor(this.win).delay(() => {
+        this.attemptToRenderCreative().then(() => {
```

Comment:
I believe this will need to be wrapped in a vsync mutate given you're modifying outside of layoutCallback.  This brings me to another point which is that we shouldn't actually render if the slot is outside of the renderOutsideViewport value otherwise we'll allow JS to execute unnecessarily and hurt the pub's active view rate.  This makes me think that we should re-evaluate if we could trigger the lifecycle to re-initate layoutCallback and destroy the previous frame/load the new one at that time.